src/server/bg_services/namespace_monitor.js

```diff
@@ -206,20 +208,41 @@ class NamespaceMonitor {
     }
 
     async test_nsfs_resource(nsr) {
+        dbg.log1('test_nsfs_resource: (name, namespace_store, nsfs_config):',
+            nsr.name, nsr.namespace_store, nsr.nsfs_config);
+        const missing_root_path_err = 'MissingRootPathError';
+        let entries;
+        let stat;
+        let error_details;
         try {
             const fs_context = {
                 backend: nsr.nsfs_config.fs_backend || '',
                 warn_threshold_ms: config.NSFS_WARN_THRESHOLD_MS,
             };
-            await nb_native().fs.readdir(fs_context, nsr.nsfs_config.fs_root_path);
-            const stat = await nb_native().fs.stat(fs_context, nsr.nsfs_config.fs_root_path);
+            entries = await nb_native().fs.readdir(fs_context, nsr.nsfs_config.fs_root_path);
+            stat = await nb_native().fs.stat(fs_context, nsr.nsfs_config.fs_root_path);
             //In the event of deleting the nsr.nsfs_config.fs_root_path in the FS side, 
-            // The number of link will be 0, then we will throw ENOENT which translate to STORAGE_NOT_EXIST
+            // The number of link will be 0, then we will throw an error which translate to STORAGE_NOT_EXIST
             if (stat.nlink === 0) {
+                error_details = 'stat.nlink === 0';
+                dbg.log1('test_nsfs_resource: the number of links is 0');
                 throw Object.assign(new Error('FS root path has no links'), { code: 'ENOENT' });
             }
         } catch (err) {
-            dbg.log1('test_nsfs_resource: got error:', err, nsr.nsfs_config);
+            if (_.isUndefined(entries)) {
+                error_details = 'nb_native().fs.readdir';
+            } else if (_.isUndefined(stat)) {
+                error_details = 'nb_native().fs.stat';
+            }
+            dbg.error('test_nsfs_resource: got error:', err, nsr.nsfs_config, 'origin of error:', error_details);
+            // we change the code to control the mapping in pool server when calc the namespace mode
+            // it can happen if (1) stat/readdir threw ENOENT or (2) the number of links is 0 
+            if (err.code === 'ENOENT') {
+                throw Object.assign(err, { code: missing_root_path_err });
```

Comment:
you can use here also  S3Error.NoSuchBucket.code, instead of the new missing root error